scalableminds/webknossos
# Add bounding box tool

### URL of deployed dev instance (used for testing):
- https://addboundingboxtool.webknossos.xyz

### Steps to test:
- Open a tracing
- Use the context menu. There should be an option for creating a bounding box. Each created bounding box should have bounds according to what the user can see. If the bounding box is created using the context menu, its center will be shifted to the point where the context menu was opened.
- Then select the bounding box tool. Each visible edge in the plane viewports should be resizable. If the cursor is near a corner, the edges of both corners should be resized. While resizing the bounds of the bounding box the edges snap automatically to full integer positions. 
- When using the bounding box tool the "create cell" button should change to a "create bounding box" button that creates a bounding box in the center of the plane viewports.
- When the bounding box tool is active and the context menu is opened on a bounding box, then the menu should have additional menu items that allow changing the color, hiding and renaming.
- The bounding box tab should still work as it did before. The onliest difference is, that a newly created bounding box via the plus button at the bottom of the bounding box list is created with the same bounds as it would be done with the "create bounding box" button in the toolbar.
- Additionally, there is another button for each bounding box in the list, that upon clicking changes the flycam's position to the center of the bounding box.
- The undo/redo feature of wk should now support resizing, color-changing, renaming, and so on of bounding boxes.
- Test what happens when the arbitrary mode is opened. The bounding box tool should automatically deactivate.
- Test what happens when the dataset only has two dimensions. Are the bounds of the created bounding boxes correct?

### Issues:
- fixes #5185, 5784

### Note:
- I am still trying to fix the tests

------
